bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,529 @@
+                    if (success) {
+                        logout();
+                        Data data = gson.fromJson(resp.data.toString(), Data.class);
+                        Dp dp = gson.fromJson(data.dp.toString(), Dp.class);
+                        Datapoints[] datapoints = gson.fromJson(dp.datapoints, Datapoints[].class);
+                        for (Datapoints datapoint : datapoints) {
+                            Descr descr = gson.fromJson(datapoint.descr, Descr.class);
+                            String channelId = "";
+                            String itemType = "String";
+                            switch (datapoint.uid) {
+                                case 2:
+                                    List<String> opModes = new ArrayList<>();
+                                    for (String modString : descr.states) {
+                                        switch (modString) {
+                                            case "0":
+                                                opModes.add("AUTO");
+                                                break;
+                                            case "1":
+                                                opModes.add("HEAT");
+                                                break;
+                                            case "2":
+                                                opModes.add("DRY");
+                                                break;
+                                            case "3":
+                                                opModes.add("FAN");
+                                                break;
+                                            case "4":
+                                                opModes.add("COOL");
+                                                break;
+                                        }
+                                    }
+                                    properties.put("Supported modes", opModes.toString());
+                                    updateProperties(properties);
+                                    channelId = CHANNEL_TYPE_MODE;
+                                    addChannel(channelId, itemType, opModes);
+                                    break;
+                                case 4:
+                                    List<String> fanLevels = new ArrayList<>();
+                                    for (String fanString : descr.states) {
+                                        if ("AUTO".contentEquals(fanString)) {
+                                            fanLevels.add("AUTO");
+                                        } else {
+                                            fanLevels.add(fanString);
+                                        }
+                                    }
+                                    properties.put("Supported fan levels", fanLevels.toString());
+                                    updateProperties(properties);
```

Comment:
Instead of calling `updateProperties` multiple times in this loop, it would be better if you only called it after you exited the loop. In fact, you might consider only calling `updateProperties` after both `populateProperties()` and this method are called as part of initialization.

Reply:
Done